token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -73,9 +79,6 @@ fn process_initialize_mint(
     confidential_transfer_mint.authority = *authority;
     confidential_transfer_mint.auto_approve_new_accounts = auto_approve_new_account;
     confidential_transfer_mint.auditor_encryption_pubkey = *auditor_encryption_pubkey;
-    confidential_transfer_mint.withdraw_withheld_authority_encryption_pubkey =
-        *withdraw_withheld_authority_encryption_pubkey;
-    confidential_transfer_mint.withheld_amount = EncryptedWithheldAmount::zeroed();
```

Comment:
Yes, I think if we do go this route, then this would be a really clean way to implement it since `instruction::initialize_mint` always happens last. We do have something roughly [similar](https://github.com/solana-labs/solana-program-library/blob/master/token/program-2022/src/extension/mod.rs#L779) for accounts, so a function that checks consistency for mints won't feel too out of place I think.

> * `ConfidentialTransferFeeConfig` without _both_ `ConfidentialTransferMint` and `TransferFeeConfig`
> * `ConfidentialTransferMint` and `TransferFeeConfig` without `ConfidentialTransferFeeConfig`

I can confirm these two checks filters all the valid/invalid extension configurations 👍